src/SIM/SIMmodel.h

```diff
@@ -0,0 +1,236 @@
+// $Id$
+//==============================================================================
+//!
+//! \file SIMmodel.h
+//!
+//! \date Sep 24 2016
+//!
+//! \author Knut Morten Okstad / SINTEF
+//!
+//! \brief Sub-class with functionality for model input and setup.
+//!
+//==============================================================================
+
+#ifndef _SIM_MODEL_H
+#define _SIM_MODEL_H
+
+#include "SIMbase.h"
+#include "TopologySet.h"
+
+class ModelGenerator;
+
+namespace LR { struct RefineData; }
+
+
+/*!
+  \brief Sub-class with functionality for model input and setup.
+  \details This class extends the SIMbase class with some added functionalities
+  for reading model definitions from input file. This also includes reading
+  initial conditions from HDF5-files and adaptive refinement. These items
+  are put in a separate sub-class to hide them away from the SIMbase class,
+  which contains the actual simulation driver (i.e., the computational methods).
+*/
+
+class SIMmodel : public SIMbase
+{
+public:
+  //! \brief Struct holding information about an initial condition.
+  struct ICInfo
+  {
+    int file_level; //!< The time level for the field in the file
+    int geo_level;  //!< The time level for the geometry in the file
+    char basis;     //!< The basis to inject field into (for mixed)
```

Comment:
either default this to 1, or keep https://github.com/OPM/IFEM/blob/master/src/SIM/SIMbase.C#L476  breaks some tests otherwise.